src/main/java/org/folio/orders/utils/HelperUtils.java

```diff
@@ -298,6 +298,10 @@ public static String getEndpointWithQuery(String query, Logger logger) {
     }
 
     List<Error> errors = new ArrayList<>();
+    
+	  if (compositeOrder.getVendor() == null || StringUtils.isEmpty(compositeOrder.getVendor())) {
```

Comment:
Agree, this can be removed `StringUtils.isEmpty(compositeOrder.getVendor())`. null should stay as is because if the vendor field does not exist then it will return null?

Reply:
the RMB @validate notation will check both for existence and the pattern, so we need not check for it again at all.

Reply:
In that case, it will not throw 422 and the test will fail. So, I think the null check is needed.

Reply:
I have pushed new changes and leveraging existing validation logic for 422 only.

Reply:
I think Kruthi is right - if vendor is missing, validation will fail with a 422 before any of our mod-orders code is called.  We typically only need to do these sorts of checks in the code when fields are required in some but not all situations...   For example if createInventory is such that a holding record should be created, we need a location, but if it's set to create nothing in inventory, or just instances, location is optional.  It's impossible to capture these rules in the json schema.  However, when a field is ALWAYS required, we can easily specify that in the schema and as Kruthi says use the @validate annotation to tell RMB to check that the request body validates against the schema.

Reply:
Good to know. Thanks, @craigmcnally @KVupp

Reply:
Good catch @KVupp .  I clearly wasn't paying attention.  This is why we need two approvals :)